nrf_rpc/nrf_rpc.c

```diff
@@ -1144,20 +1162,66 @@ int nrf_rpc_init(nrf_rpc_err_handler_t err_handler)
 		return err;
 	}
 
-	for (i = 0; i < CONFIG_NRF_RPC_CMD_CTX_POOL_SIZE; i++) {
+	for (size_t i = 0; i < CONFIG_NRF_RPC_CMD_CTX_POOL_SIZE; i++) {
 		cmd_ctx_pool[i].id = i;
 		err = nrf_rpc_os_msg_init(&cmd_ctx_pool[i].recv_msg);
 		if (err < 0) {
 			return err;
 		}
 	}
 
-	err = transport_init(receive_handler);
+	global_err_handler = default_err_handler;
+
+	is_initialized = true;
+
+	return 0;
+}
+
+int nrf_rpc_init_group(const struct nrf_rpc_group *group)
+{
+	int err = nrf_rpc_prepare_init();
```

Comment:
It seems that this code is not thread-safe. Is there an assumption that `nrf_rpc_init_group` must be called from the main thread only? If so, I think we should document it. Otherwise, `nrf_rpc_prepare_init` may end up being executed from two threads.

Reply:
You are right that this  is not thread safe, but is it not the case that the whole nrf_pc is not thread safe? The variable `is_initialized` was static before, I didn't change that. I am not too familiar with the library but it I see many static objects being manipulated so I thought that it is not thread safe. Am I missing something?